src/SurveyService.Validation/Question/CreateQuestionRequestValidator.cs

```diff
@@ -0,0 +1,56 @@
+﻿using FluentValidation;
+using LT.DigitalOffice.SurveyService.Data.Interfaces;
+using LT.DigitalOffice.SurveyService.Models.Dto.Requests.Question;
+using LT.DigitalOffice.SurveyService.Models.Dto.Requests.Question.Filters;
+using LT.DigitalOffice.SurveyService.Validation.Option.Interfaces;
+using LT.DigitalOffice.SurveyService.Validation.Question.Interfaces;
+using System;
+
+namespace LT.DigitalOffice.SurveyService.Validation.Question;
+
+public class CreateQuestionRequestValidator : AbstractValidator<CreateQuestionRequest>, ICreateQuestionRequestValidator
+{
+  public CreateQuestionRequestValidator(
+    IQuestionRepository _questionRepository
+    )
+  {
+    RuleFor(q => q.Content)
+      .MaximumLength(300)
+      .WithMessage("Content is too long.");
+
+    When(q => q.Deadline.HasValue, () =>
+    {
+      RuleFor(q => q.Deadline)
+        .Must(v => v > new DateTime(v.Value.Year, v.Value.Month, v.Value.Day + 1,
+                                    v.Value.Hour, v.Value.Minute, v.Value.Millisecond))
+        .WithMessage("The deadline must be at least 24 hours from now.");
+    });
+
+    When(q => q.GroupId.HasValue, () =>
+    {
+      RuleFor(q => q)
+        .MustAsync(async (q, _) => (await _questionRepository.GetPropertiesAsync(new GetQuestionPropertiesFilter() { GroupId = q.GroupId })).Deadline == q.Deadline)
+        .WithMessage("The deadlines of questions in the group are not equal.");
+
+      RuleFor(q => q)
+        .MustAsync(async (q, _) => (await _questionRepository.GetPropertiesAsync(new GetQuestionPropertiesFilter() { GroupId = q.GroupId })).HasRealTimeResult == q.HasRealTimeResult)
+        .WithMessage("The conditions for displaying the results of the questions are different.");
+    });
+
+    When(q => !q.HasMultipleChoice && q.HasCustomOptions, () =>
+    {
+      RuleFor(q => q.Options)
+        .NotEmpty().WithMessage("No options in the question.");
+    });
+
+    When(q => !q.HasCustomOptions, () =>
```

Comment:
For me it seems like not a very good idea to do checks on options in question validator, it is CrateOptionRequestValidator responsobility.

Reply:
We discussed that we would make another validator for the options in the question

Reply:
done